pkg/codefresh/git-source.go

```diff
@@ -0,0 +1,94 @@
+package codefresh
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/codefresh-io/go-sdk/pkg/codefresh/model"
+)
+
+type (
+	IGitSourceAPI interface {
+		List(runtimeName string) ([]model.GitSource, error)
+	}
+
+	gitSource struct {
+		codefresh *codefresh
+	}
+
+	graphQlGitSourcesListResponse struct {
+		Data struct {
+			GitSources model.GitSourcePage
+		}
+		Errors []graphqlError
+	}
+)
+
+func newGitSourceAPI(codefresh *codefresh) IGitSourceAPI {
+	return &gitSource{codefresh: codefresh}
+}
+
+func (g *gitSource) List(runtimeName string) ([]model.GitSource, error) {
+	jsonData := map[string]interface{}{
+		"query": `{
```

Comment:
query must return gitSources by `runtime` parameter. Not all the gitSources in the system.